Magento_Theme/web/js/custom-modal.js

```diff
@@ -0,0 +1,44 @@
+define(['jquery', 'jquery-ui-modules/widget', 'mage/cookies'], function ($) {
+  $.widget('vaimo.modalWidget', {
+    options: {
+      modalOpenClass: 'modal',
+      modalNotOpenClass: 'not-open-modal',
+      modalShadowClass: 'modal-shadow',
+    },
+
+    _create: function () {
+      if (!$.mage.cookies.get('modalClosed')) {
+        this.element.css('display', 'block');
+        this.addModalShadow();
+      }
+      this._on({
+        'click #close-modal-icon': this.closeModal,
```

Comment:
add `.bind(this)` in the end of every callback.
read more about object context and `this`
here:
- https://javascript.info/object-methods
- https://javascript.info/bind
- https://javascript.info/arrow-functions